sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java

```diff
@@ -0,0 +1,535 @@
+    @Nullable
+    protected String filter;
+    protected int numberSplit;
+
+    private Read() {}
+
+    private Read(String uri, String database, String collection, String filter, int numberSplit) {
+      this.uri = uri;
+      this.database = database;
+      this.collection = collection;
+      this.filter = filter;
+      this.numberSplit = numberSplit;
+    }
+
+    @Override
+    public PCollection<String> apply(PBegin input) {
+      org.apache.beam.sdk.io.Read.Bounded bounded = org.apache.beam.sdk.io.
+          Read.from(createSource());
+      PTransform<PInput, PCollection<String>> transform = bounded;
+      return input.getPipeline().apply(transform);
+    }
+
+    /**
+     * Creates a {@link BoundedSource} with the configuration in {@link Read}.
+     */
+    @VisibleForTesting
+    BoundedSource createSource() {
+      return new BoundedMongoDbSource(uri, database, collection, filter, numberSplit);
+    }
+
+    @Override
+    public void validate(PBegin input) {
+      Preconditions.checkNotNull(uri, "uri");
+      Preconditions.checkNotNull(database, "database");
+      Preconditions.checkNotNull(collection, "collection");
+    }
+
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+      super.populateDisplayData(builder);
+
+      builder.addIfNotNull(DisplayData.item("uri", uri));
+      builder.addIfNotNull(DisplayData.item("database", database));
+      builder.addIfNotNull(DisplayData.item("collection", collection));
+      builder.addIfNotNull(DisplayData.item("filter", filter));
+      builder.addIfNotNull(DisplayData.item("numberSplit", numberSplit));
+    }
+
+  }
+
+  private static class BoundedMongoDbSource extends BoundedSource {
+
+    private String uri;
+    private String database;
+    private String collection;
+    @Nullable
+    private String filter;
+    private int numberSplit;
+    private Long avgSize;
+
+    public BoundedMongoDbSource(String uri, String database, String collection, String filter,
+                                int numberSplit) {
+      this.uri = uri;
+      this.database = database;
+      this.collection = collection;
+      this.filter = filter;
+      this.numberSplit = numberSplit;
+    }
+
+    @Override
+    public Coder getDefaultOutputCoder() {
+      return SerializableCoder.of(String.class);
+    }
+
+    @Override
+    public void validate() {
+      Preconditions.checkNotNull(uri, "uri");
+      Preconditions.checkNotNull(database, "database");
+      Preconditions.checkNotNull(collection, "collection");
+    }
+
+    @Override
+    public boolean producesSortedKeys(PipelineOptions options) {
+      return false;
+    }
+
+    @Override
+    public BoundedReader createReader(PipelineOptions options) {
+      return new BoundedMongoDbReader(uri, database, collection, filter);
+    }
+
+    @Override
+    public long getEstimatedSizeBytes(PipelineOptions pipelineOptions) {
+      Long estimatedByteSize = 0L;
```

Comment:
Would be better to use a regular "long" - Long hints that these might be null, but that's not the case. Applies to all other usages of Long in this PR.